src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -112,21 +112,24 @@ public function onKernelResponse(FilterResponseEvent $event)
     protected function injectToolbar(Response $response, Request $request)
     {
         $content = $response->getContent();
-        $pos = strripos($content, '</body>');
-
-        if (false !== $pos) {
-            $toolbar = "\n".str_replace("\n", '', $this->twig->render(
-                '@WebProfiler/Profiler/toolbar_js.html.twig',
-                array(
-                    'position' => $this->position,
-                    'excluded_ajax_paths' => $this->excludedAjaxPaths,
-                    'token' => $response->headers->get('X-Debug-Token'),
-                    'request' => $request,
-                )
-            ))."\n";
-            $content = substr($content, 0, $pos).$toolbar.substr($content, $pos);
-            $response->setContent($content);
+        $injectToolbarAt = strripos($content, '</body>') ?: strripos($content, '</html>');
+        $canInjectToolbar = $injectToolbarAt || stripos($content, '<title>');
```

Comment:
Currently, we err on the side of not injecting the toolbar when it could be (but you can still access the data by looking for recent profiles in the profiler). Your PR allows to break the response in some case by injecting the toolbar when it should not.

Is it possible to write a valid HTML full page without a ``</html>`` (or without a ``<html`` ? If no, these should be used for the detecting instead IMO (``</html>`` is already used by your change btw)